doc/kci_rootfs.md

```diff
@@ -0,0 +1,156 @@
+
+   ```
+   sudo docker run -itd -v $(pwd)/kernelci-core:/kernelci-core --device /dev/kvm --privileged kernelci/staging-debos
+   sudo docker exec -it <container_id> bash
+   cd /kernelci-core/ 
+   ```
+4. Now to check if everything works, type `./kci_rootfs --help` it should produce below help message.
+
+    ```
+    usage: kci_rootfs [-h] [--yaml-configs YAML_CONFIGS]
+
+                      {validate,list_configs,list_variants,build,upload} ...
+
+    optional arguments:
+
+      -h, --help            show this help message and exit
+
+      --yaml-configs YAML_CONFIGS
+
+                            Path to the YAML configs file
+    Commands:
+
+      {validate,list_configs,list_variants,build,upload}
+
+                            List of available commands
+
+        validate            Validate the rootfs YAML configuration
+
+        list_configs        List all rootfs config names
+
+        list_variants       List all rootfs variants
+
+        build               Build a rootfs image
+
+        upload              Upload a rootfs image
+    ```
+5. To list available rootfs configuration, we can use `list_configs` option.
+
+    ```
+    $ ./kci_rootfs list_configs       
+
+    buster
+
+    buster-v4l2
+
+    buster-igt
+    ```
+   By default `kci_rootfs` reads entries from `rootfs-configs.yaml`. This file acts as a rootfs config file.
+
+6. Lets list all available rootfs images using `./kci_rootfs list_variants`. It should show existing
+rootfs name along with its architecture. 
+
+    ```
+    buster amd64
+    buster arm64
+    ```
+
+    It also comes with couple of options `--config` and `--arch` to filter the results based on config name or arch type.
+    
+    ```
+    $ ./kci_rootfs list_variants --config buster --arch i386
+    buster i386
+
+    $ ./kci_rootfs list_variants --config buster 
+    buster amd64
+    buster arm64
+
+    $ ./kci_rootfs list_variants --arch amd64
+    buster amd64
+    buster-v4l2 amd64
+    buster-igt amd64
+    ```
+7. Now its time to re-build existing rootfs image using `kci_rootfs build`. It takes three arguments,
+`--config` refers to rootfs name which we want to build. `--data-path` points to debos files location.
+`--arch` refers to CPU arch we want to build. For example, to build a buster rootfs image for i386, 
+we can run
+
+    ```
+    ./kci_rootfs build --config buster --data-path jenkins/debian/debos --arch i386
+
+    ```
+   depending on your network speed, this will take sometime to complete. 
+
+8. If build is successful you should see message like
+
+    ```
+    cd ${ROOTDIR} ; find -H  |  cpio -H newc -v -o | gzip -c - > ${ARTIFACTDIR}/buster/i386/rootfs.cpio.gz | ./build_info.json
+    cd ${ROOTDIR} ; find -H  |  cpio -H newc -v -o | gzip -c - > ${ARTIFACTDIR}/buster/i386/rootfs.cpio.gz | 79539 blocks
+    Powering off.
+    ==== Recipe done ====
+    ```
+    Finally newly built rootfs images can be found at `jenkins/debian/debos/buster/i386/`
```

Comment:
Maybe we should mention that the directories are created relatively to the `--data-path` value.